### PR TITLE
Filter out future purge dates resources

### PR DESCRIPTION
**Pull request by weshaggard**
Most (>8k) of the deleted key vaults have a future purge date and we fail to delete them and we end up timing out so I'm filtering so that we only attempt to delete ones with past due purge dates.

### CODE REVIEW

**Comment by weshaggard** [comment]
@heaths I'm merging this for now to give the cleanup automation a change to finish as it keeps timing out currently and even locally I seem to keep hitting hangs or timeouts on certain resources. If you know of a better way to filter what we can and cannot delete yet let me know.

File: eng/scripts/live-test-resource-cleanup.ps1 [hunk 1]
```diff
@@ -86,6 +85,16 @@ foreach ($rg in $toDelete)
   }
 }
 
+# Get purgeable resources already in a deleted state coerced into a collection even if empty.
+$purgeableResources = Get-PurgeableResources
+$allPurgeCount = $purgeableResources.Count
+
+# Filter down to the ones that we can actually perge.
```
Comment (heaths) [inline eng/scripts/live-test-resource-cleanup.ps1:92]: "purge"

/cc @weshaggard

File: eng/scripts/live-test-resource-cleanup.ps1 [hunk 2]
```diff
@@ -86,6 +85,16 @@ foreach ($rg in $toDelete)
...
+$purgeableResources = $purgeableResources.Where({ $purgeDate = $_.ScheduledPurgeDate -as [DateTime]; (!$purgeDate -or $now -gt $purgeDate) })
```
Comment (heaths) [inline eng/scripts/live-test-resource-cleanup.ps1:93]: This is now how you filter them. You need to check EnableSoftDelete. You've also not purged the vaults early enough. The default is 90d, so you're effectively leaving hundreds of vaults un-purged.

Comment (heaths) [reply eng/scripts/live-test-resource-cleanup.ps1:93]: There's no point in purging vaults in the future. Once the purge date is hit, they are purged automatically.

**Comment by heaths** [comment]
> Most (>8k) of the deleted key vaults have a future purge date and we fail to delete them and we end up timing out so I'm filtering so that we only attempt to delete ones with past due purge dates.

That's not why some fail to delete. The ones that fail to delete is because EnableSoftDelete is null or false. As I explained in discussions with my PR, the default for EnableSoftDelete *was* `false`, but now is `true` and cannot be disabled anymore for new vaults. That's why I try to delete them anyway.

Vaults and managed HSMs are automatically deleted on the purge dates. That's what that property is for. By not deleting vaults and managed HSMs until after their purge date, you've effectively obviated the whole point of this clean-up exercise.

**Pull request merged**